kuma/javascript/src/payments/index.jsx

```diff
@@ -28,15 +33,25 @@ export default function PaymentsLandingPage() {
         terms: `/${locale}/payments/terms`
     };
 
+    let showSubscriptionForm = false;
+    // if the subscription banner is not enabled,
+    // do not show the <SubscriptionForm />
+    if (userData && userData.waffle.flags['subscription_banner']) {
+        showSubscriptionForm = true;
+    }
+
     return (
         <>
             <A11yNav />
             <Header />
-            <SubHeader
-                title="Become a monthly supporter"
-                description="Support MDN with a $5 monthly subscription and get back more of the knowledge and tools you rely on for when your work has to work."
-                columnWidth="7"
-            />
+            <div className="subscriptions subheader-container">
+                <SubHeader
+                    title="Become a monthly supporter"
+                    description="Support MDN with a $5 monthly subscription and get back more of the knowledge and tools you rely on for when your work has to work."
+                    columnWidth="7"
+                />
+                <SubscriptionForm showSubscriptionForm={showSubscriptionForm} />
```

Comment:
@Gregoor not sure is this is the best way to do this. Also not sure if it would be better to instead use `useState` instead of a variable.

With the current set-up, things work as expected but, I do see the following in the developer console:

```
Warning: Did not expect server HTML to contain a <div> in <div>.
```

Note: I am testing this after running `npm test` so the SSR server is not running.

What I also notice is that if I add a `console.log` at line 16 of `subscription-form.jsx` it is logged 3 times on a single page load. Perhaps this is normal, but I thought I would call it out.

Reply:
Also, if it _is_ better to use `useState`, I am curious whether it is a case of, "in these kinds of cases, _always_ use useState"

Reply:
Since the component doesn't need to do anything when not shown (apart from not showing up), a more React-y way of doing this would be:

```jsx
return (
  <stuff>
  {showSubscriptionForm && <SubscriptionForm />}
  </stuff>
);
```

Then you can also forego having `showSubscriptionForm` as a prop.

Wrt client and server rendering differing as well as render running multiple times, will look at it in action, but here's some wild guesses:
- SSR is at times quite stubborn for me locally with rendering new changes, it might also still be running on an old version of the code for you
- render does need to be re-run at least once, given that userData only comes in later (after the first render)

Reply:
> 
> 
> Also, if it _is_ better to use `useState`, I am curious whether it is a case of, "in these kinds of cases, _always_ use useState"

`useState` would not be better here, since the value of the variable is wholly dependdent on the context's state and can be computed from it with no larger costs. If it was something computationally heavy it might be justified to use state with a function initializer, i.e.
```js
const showSubscriptionForm = useState(() => someHeavyComputation());
```

Reply:
> Since the component doesn't need to do anything when not shown (apart from not showing up), a more React-y way of doing this would be:

I completely forgot that I could do that! Thanks Gregor!

Reply:
Perhaps you can send a PR to update the docs here :)
https://reactjs.org/docs/conditional-rendering.html#preventing-component-from-rendering